ls1intum/Artemis
# `Programming exercises`: Update template dependencies for Java exercises

### Checklist
#### General
#### Server
#### Changes affecting Programming Exercises

### Motivation and Context
This PR updates multiple dependencies used in the templates to their newest version.

Related PR for the docker-container: https://github.com/ls1intum/artemis-maven-docker/pull/97

### Description
Updates Plugin versions and Gradle to [7.5.1](https://gradle.org/releases/).

Note: Changes to gradlew.file, gradlew.bat and gradle-wrapper.jar are automatically generated.

### Steps for Testing
Prerequisites:
- 1 Instructor
- 1 Student

1. Create new Programming Exercises (Gradle and Maven)
2. Check that the template works correctly (compilation / running tests / running SCA still possible)

### Review Progress

#### Code Review
#### Manual Tests
